### PR TITLE
Add cuda option to inference on 1 GPU

### PR DESCRIPTION
**Pull request by OvsiienkoVladyslava**
Add option to run code on GPU(1 GPU) or CPU

### CODE REVIEW

File: pytorch_object_detection/README.md [hunk 1]
```diff
@@ -15,17 +15,21 @@ Source: https://pytorch.org/vision/stable/models.html
...
+If you want to run your code using GPU you need to install [cuda version 11.7](https://developer.nvidia.com/cuda-11-7-0-download-archive?target_os=Windows&target_arch=x86_64&target_version=11) (to check if you've installed it run in cmd `nvcc --version`)
+
 3. Run in cmd or your IDE with parameters:  
 
 _--source-folder_ - path to folder with images, e.g. "D:\\project\\test_data"<br />
 _--model-name_ - model name to use, available: 'Faster R-CNN, RetinaNet'<br />
 _--score-threshold_ - min confidence threshold of prediction, e.g 0.8<br />
-_--save-folder-path_ - path to folder where to save predicted labels, boxes, scores, images, e.g. "./test_results"
+_--save-folder-path_ - path to folder where to save predicted labels, boxes, scores, images, e.g. "./test_results"<br />
+_--save-folder-path_ - if to run code on GPU if cuda is available otherwise run on CPU
```
Comment (AlexShashko) [inline pytorch_object_detection/README.md]: Maybe "--run_on_gpu" flag name?

File: pytorch_object_detection/README.md [hunk 2]
```diff
@@ -15,17 +15,21 @@ Source: https://pytorch.org/vision/stable/models.html
...
 
 4. Example
 
 **Console input:** <br />
-`$ python run.py --source-folder "D:\\computer vision\\pytorch_object_detection\\test_data" --model-name "RetinaNet" --score-threshold 0.8 --save-folder-path "./test_results"`<br />
+`$ python run.py --source-folder "D:\\computer vision\\pytorch_object_detection\\test_data" --model-name "RetinaNet" --score-threshold 0.8 --save-folder-path "./test_results" --run_on_gpu
+True`<br />
```
Comment (AlexShashko) [inline pytorch_object_detection/README.md]: Remove True, keys can have binary logic.

File: pytorch_object_detection/detection_models/__init__.py [hunk 3]
```diff
@@ -17,15 +17,20 @@ class DetectionPipeline(ABC):
...
-    def __init__(self):
+    def __init__(self, run_on_gpu: bool = True):
         """
         In __init__ func of child class it is needed to implement 2 abstract methods:
         1. define weights of pretrained model
         2. define model with these weights
+
+        :param run_on_gpu: if to run code on GPU
         """
+        # Select device - if GPU available and needed and print information about used device
```
Comment (AlexShashko) [inline pytorch_object_detection/detection_models/__init__.py]:
```suggestion
        # Select device - if GPU available, requested and print information about used device
```

File: pytorch_object_detection/detection_models/__init__.py [hunk 4]
```diff
@@ -68,7 +92,7 @@ def draw_boxes(imgs: List[Tensor], boxes: List[list], labels: List[List[str]]) -
         """
         images_with_boxes = []
         for ind, img in enumerate(imgs):
-            predicted_boxes = torch.Tensor(boxes[ind])
+            predicted_boxes = torch.Tensor(boxes[ind]).to(self.device)
```
Comment (AlexShashko) [inline pytorch_object_detection/detection_models/__init__.py]: Why did you send boxes to GPU before drawing?

File: pytorch_object_detection/detection_models/run.py [hunk 5]
```diff
@@ -42,5 +44,10 @@ def process_console_run(source_folder: str, model_name: str, score_threshold: fl
         default="./test_results",
         help="path to folder where to save predicted labels, boxes, scores, images," ' e.g. "./test_results"',
     )
+    parser.add_argument(
+        "--run_on_gpu", type=bool, default=True, help="if to run code on GPU if cuda is available "
```
Comment (AlexShashko) [inline pytorch_object_detection/detection_models/run.py]: Implement binary logic for the flag.

File: pytorch_object_detection/detection_models/__init__.py [hunk 6]
```diff
@@ -17,15 +17,20 @@ class DetectionPipeline(ABC):
     Class defining basic functionality to run pretrained object detection model on image or batch of images.
     """
 
-    def __init__(self):
+    def __init__(self, run_on_gpu: bool = True):
```
Comment (AlexShashko) [inline pytorch_object_detection/detection_models/__init__.py:30]: Is this parameter implemented in child classes?

File: pytorch_object_detection/detection_models/__init__.py (hunk 4 above)
Comment (OvsiienkoVladyslava) [reply pytorch_object_detection/detection_models/__init__.py]: let it be

File: pytorch_object_detection/detection_models/__init__.py (hunk 6 above)
Comment (OvsiienkoVladyslava) [reply pytorch_object_detection/detection_models/__init__.py:30]: they do not implement init

**Review by AlexShashko — APPROVED** [review]

**Pull request merged**